pkg/services/repo_host/github.go

```diff
@@ -0,0 +1,77 @@
+package repo_host
+
+import (
+	"context"
+
+	"github.com/google/go-github/v35/github"
+	"golang.org/x/oauth2"
+)
+
+type GithubRepoHost struct {
+	client       *github.Client
+	organization string
+}
+
+var _ RepoHostService = &GithubRepoHost{}
+
+func NewGithubRepoHost(ctx context.Context, organization string, token string, url string) (*GithubRepoHost, error) {
+	var ts oauth2.TokenSource
+	if token != "" {
+		ts = oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		)
+	}
+	httpClient := oauth2.NewClient(ctx, ts)
+	var client *github.Client
+	if url == "" {
+		client = github.NewClient(httpClient)
+	} else {
+		var err error
+		client, err = github.NewEnterpriseClient(url, url, httpClient)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &GithubRepoHost{client: client, organization: organization}, nil
+}
+
+func (g *GithubRepoHost) ListRepos(ctx context.Context) ([]*HostedRepo, error) {
+	opt := &github.RepositoryListByOrgOptions{
+		ListOptions: github.ListOptions{PerPage: 100},
+	}
+	repos := []*HostedRepo{}
+	for {
+		githubRepos, resp, err := g.client.Repositories.ListByOrg(ctx, g.organization, opt)
+		if err != nil {
+			return nil, err
+		}
+		for _, githubRepo := range githubRepos {
+			repos = append(repos, &HostedRepo{
+				Organization: githubRepo.Owner.GetName(),
+				Repository:   githubRepo.GetName(),
+				URL:          githubRepo.GetSSHURL(), // TODO Config flag for CloneURL (i.e. https://)?
```

Comment:
This always uses the SSH URL, some users might prefer the HTTPS URL for cloning (eg. using token authentication instead of SSH keys). I think this is the more common use case with Argo so it's probably the right default but Argo supports a lot of auth modes (basic auth, token, github app, TLS auth which I don't think GHE supports anyway). It should be configurable in the `github: {}` struct with maybe an overall default in the generator since SSH vs HTTPS is common for many providers.